### ISSUE TITLE
[TvOS] : React Native assets registry version mismatch

### ISSUE THREAD

**Issue by stdiodavid** [expo/expo#29607]
### Summary

1.npx create-expo-app MyTVProject -e with-tv
2.cd MyTVProject && export EXPO_TV=1 && npx expo prebuild -p ios && npm run ios

tv screen 1
<img width="603" alt="image" src="https://github.com/expo/expo/assets/144206414/f92a8d58-2acc-43af-8c85-68ab55172724">

tv screen 2
<img width="532" alt="image" src="https://github.com/expo/expo/assets/144206414/678f52b9-d6a9-4509-9b7b-a7dd6e4b7ed6">

### What platform(s) does this occur on?

iOS

### SDK Version

50

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.20.3 - /usr/local/opt/node@18/bin/node
      Yarn: 1.22.22 - /usr/local/bin/yarn
      npm: 10.7.0 - /usr/local/opt/node@18/bin/npm
      Watchman: 2024.05.06.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.8 => 51.0.11
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native-tvos:  0.74.1-0
    Expo Workflow: bare

### Minimal reproducible example

replace app.tsx
```
import { StyleSheet, Text, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';
export default function App() {
  return (
    <View style={styles.container}>
      <Ionicons name="checkmark-circle" size={32} color="green" />
      <Text>Open up App.tsx to start working on your app!</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```

**Comment by brentvatne** [comment]
@douglowder - i believe this is related to the react-native asset registry - the version of the package changed with a react-native version bump and issues like this arise when there are multiple versions of the package. https://github.com/expo/expo/pull/29541 is a fix for this more systemtically

@stdiodavid - you can work around this by using yarn resolutions / npm overrides to force a specific version of `@react-native/assets-registry`.

**Comment by douglowder** [comment]
The TV examples (with-tv, with-router-tv) have been updated to RNTV 0.74.2-0 and the latest Expo SDK 51 packages, and the issue is resolved (only one assets-registry version appears in the dependencies). This also fixes #29600.
